src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,382 @@
+      // set original value (consisting of only whitespace) to CDF field
+      recBuilder.set(fieldName, isTypeString ? fieldValStr : null);
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Processes field native value according to their type based on Schema Simple
+   * Type or Logical Type
+   *
+   * @param nonNullSchema
+   * @param recBuilder
+   * @param encodedFieldName
+   * @param fieldValStr
+   * @throws IOException
+   */
+  private void processValue(Schema nonNullSchema, StructuredRecord.Builder recBuilder, String encodedFieldName,
+                            String fieldValStr)
+    throws IOException {
+
+    try {
+      if (nonNullSchema.getLogicalType() != null) {
+        processLogicalTypeVal(nonNullSchema, recBuilder, encodedFieldName, fieldValStr.trim());
+      } else {
+        processTypeVal(nonNullSchema.getType(), recBuilder, encodedFieldName, fieldValStr);
+      }
+    } catch (Exception e) {
+      handleConversionException(nonNullSchema.getLogicalType() != null ? nonNullSchema.getLogicalType().toString()
+                                  : nonNullSchema.getType().toString(), encodedFieldName, fieldValStr, e);
+    }
+  }
+
+  /**
+   * Process the value for field which is mapped to a {@code Schema.LogicalType}
+   * and set into the {@code StructuredRecord.Builder}.
+   *
+   * @param nonNullSchema non nullable Schema
+   * @param recBuilder    Structured record builder
+   * @param fieldName     Ariba objects's field name (may be encoded to remove CDF
+   *                      unsupported chars)
+   * @param fieldValTrim  trimmed value corresponding to an Ariba object's field
+   *                      name
+   */
+  private void processLogicalTypeVal(Schema nonNullSchema, StructuredRecord.Builder recBuilder, String fieldName,
+                                     String fieldValTrim) {
+
+    switch (Objects.requireNonNull(nonNullSchema.getLogicalType())) {
+      case DECIMAL:
+        fieldValTrim = handleMinusAtEnd(fieldValTrim);
+        recBuilder.setDecimal(fieldName, new BigDecimal(fieldValTrim).setScale(nonNullSchema.getScale()));
+        break;
+
+      case DATE:
+        recBuilder.setDate(fieldName, getSourceSpecificDateValue(fieldValTrim));
+        break;
+
+      case TIME_MICROS:
+        recBuilder.setTime(fieldName, getSourceSpecificTimeValue(fieldValTrim));
+        break;
+
+      case TIMESTAMP_MICROS:
+        ZonedDateTime zonedDateTime = null;
+        // Check if UTCLONG string having format yyyy-MM-dd HH:mm:ss'Z' does not
+        // start with default date value part 0000
+        if (!fieldValTrim.startsWith("0000")) {
+          String parsableTimestamp = fieldValTrim.replace(' ', 'T');
+          String nanoSecPart = "0";
+          zonedDateTime = ZonedDateTime.parse(parsableTimestamp, DateTimeFormatter.ISO_DATE_TIME);
+          zonedDateTime = zonedDateTime.plus(Long.parseLong(nanoSecPart), ChronoUnit.NANOS);
```

Comment:
why do you need this? Also, no need for Long.parseLong("0"), you can just specify 0.

Reply:
Fixed